File: microservices/movie-service/src/main/java/com/project/movie/services/TheatreService.java
package com.project.movie.services;

import com.project.movie.entities.Theatre;

import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface TheatreService {
	
	public Mono<Theatre> getTheatreById(String theatreId);
	
	public Flux<Theatre> getAllTheatres();
	
	public Flux<Theatre> getTheatreByCity(String city);
	
	public Mono<Theatre> addTheatre(Theatre theatre);
	
	public Mono<Theatre> updateTheatre(Theatre theatre);
	
	public Mono<?> deleteTheatre(String theatreId);

}

File: ui/movie-booking-app/src/app/services/theatre.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { Theatre } from '../models/theatre.model';

@Injectable({
  providedIn: 'root'
})
export class TheatreService {

  _baseUrl = `${environment.baseUrl}/movie-service/theatre`;

  constructor( private http: HttpClient ) { }

  getAllTheatres(): Observable<Theatre[]> {
    return this.http.get<Theatre[]>(`${this._baseUrl}`);
  }

  getTheatreById(theatreId: string): Observable<Theatre> {
    return this.http.get<Theatre>(`${this._baseUrl}/${theatreId}`);
  }

  getTheatresByCity(city: string): Observable<Theatre[]> {
    return this.http.get<Theatre[]>(`${this._baseUrl}/city/${city}`);
  }

  addTheatre(theatre: Theatre): Observable<Theatre> {
    return this.http.post<Theatre>(`${this._baseUrl}`, theatre);
  }

  updateTheatre(theatre: Theatre): Observable<Theatre> {
    return this.http.put<Theatre>(`${this._baseUrl}`, theatre);
  }

  deleteTheatre(theatreId: string): Observable<any> {
    return this.http.delete<any>(`${this._baseUrl}/${theatreId}`);
  }
}
